### DISCUSSION TITLE
About formatting source codes

### DISCUSSION THREAD

**Discussion by cool0707** [igniterealtime/openfire-pade-plugin#296] [General]
I'm thinking of using VS code's "Beautify" plugin to format the code, since `custom_ofmeet.js` etc. doesn't have consistent formatting rules.
I'm thinking of adding a ".jsbeautifyrc" file to the project root, but we need to decide whether to use the [K&R style](https://en.wikipedia.org/wiki/Indentation_style#K&R_style) or the [Allman style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style).
Most of `custom_ofmeet.js` is written in the Allman style, but the mainstream JavaScript nowadays is K&R style.

**Comment by gjaekel** [comment]
Personally, I *hate* the Allman style. I bend myself as expected in a project. I think it's because the first serious programming language I was educated in was Pascal with it's block terminating keyword syntax.

:+1: for Allman!

I also tend to violate K&R for "one-liners" like `if (....) { some simple things }` as found in the `...settings.java`

**Reply by deleolajide** [reply]
I am not fussed about any style. I started with a K&R-ish style from my earliest days of coding Fortran on a mainframe. Somewhere between coding C, Basic and Java I picked up Allman. In general I try and maintain the coding style I find, but resort to Allman when I code something new.

**Comment by cool0707** [comment]
I'm not particularly fussy about style either, so I tend to follow the default rules of the IDE formatter.
Jitsi Meet and other libraries are also K&R, so I'll use K&R.

By the way, "Beautify" confuses `of` variable in `custom_ofmeet.js` with the `of` keyword of `for...of` statement and adds a space before and after it.
For this reason, I'm going to rename it to `ofm`, etc.

**Comment by gjaekel** [comment]
> For this reason, I'm going to rename it to ofm, etc.

There's the variable named `APP`; what's about to use `OF` (or `OFM`)?

**Reply by cool0707** [reply]
It is used in this area.
I don't particularly feel the need to add `recording` etc. as properties of `of`, but I think @deleolajide is using it with some intention.

https://github.com/igniterealtime/openfire-pade-plugin/blob/a8178598de4a04ebccd5be0597cac412f5c80e77/src/webapp/custom_ofmeet.js#L1

https://github.com/igniterealtime/openfire-pade-plugin/blob/a8178598de4a04ebccd5be0597cac412f5c80e77/src/webapp/custom_ofmeet.js#L159-L187

**Reply by deleolajide** [reply]
>There's the variable named APP; what's about to use OF (or OFM)?

APP has global scope. thatis why all caps is used. OF or OFM would be wrong because of.recording has scope only within the ofmeet module. To use it outside of the module, you would use ofmeet.recording.

The need to access ofmeet.recording externally is no more needed. In the past when it was part of the Pade chrome extension client, it was used in the background page on the window close event handler to cleanup an active recording. Now this is handled in the ofmeet module itself.

In any case, I think we should apply this later this week on next week after I release 1.3.3
